src/cargo/util/toml/mod.rs

```diff
@@ -562,6 +563,18 @@ impl TomlProfile {
         if self.strip.is_some() {
             features.require(Feature::strip())?;
         }
+
+        if let Some(codegen_backend) = &self.codegen_backend {
+            features.require(Feature::codegen_backend())?;
+            if codegen_backend.contains(|c| !c.is_ascii_alphanumeric() && c != '_') {
```

Comment:
```suggestion
            if codegen_backend.contains(|c: char| !c.is_ascii_alphanumeric() && c != '_') {
```

Sorry, that CI failure was my fault with the previous code suggestion.

Reply:
I actually only noticed the `bail!()` change. Rustc uses the existence of `.` to distinguish between builtin and external backends.

Reply:
I didn't know that. We can always expand what this allows if rustc adds a built-in backend that needs a broader set of potential inputs.